File: 5.Semester/SWK5/Uebung/Beispiele/Ue01/Shared/Calc.cs
using System.Reflection;
[assembly:AssemblyVersion("1.0.0.0")]
public class Calc
{
    protected double sum = 0;
    protected int n = 0;

    public void Add(double number)
    {
        this.sum += number;
        this.n++;
    }

    public double GetSum(){
        return this.sum;
    }
}

File: 5.Semester/SWK5/Uebung/Beispiele/UE01 03.10.18/UnmanagedCode/Calc.cpp
#include <cstdio>
#include "Calc.h"

Calc::Calc() {
	sum = 0;
	n   = 0;
}

Calc::~Calc() {
  printf("~Calc\n");
}

void Calc::Add(double number) {
	sum += number;
	n++;
}

double Calc::GetSum() {
	return sum;
}
